scripts/generate_code.py

```diff
@@ -192,6 +193,11 @@ def glob(path, pattern):
     data="monsterdata_test.json",
 )
 
+flatc(
```

Comment:
We can technically use the same script used in #7827.

During code generator interface migration, flatc with `--proto` error wasn't caught. Since we already require running `scripts/generate_code.py`, I think it's better to add this test to ensure the same error doesn't happen again.

Reply:
In a another CL or this one, I think we should migrate all the golden generation from `GenerateProtoGoldens.sh`  to `generate_code.py`. Since the bash script isn't included in our CI, this would help us catch  any breaking changes to the fbs generator in the future.

Reply:
If we were to migrate all the bash scripts in GenerateProtoGoldens.sh, I think we should do that in another PR. It would keep the work in separate PRs. And also, the PR will be smaller to review. I can work on migrating it over in once this one is submitted.